JPEGView/ImageLoadThread.cpp

```diff
@@ -424,11 +453,16 @@ void CImageLoadThread::ProcessReadWEBPRequest(CRequest * request) {
 			if (Webp_Dll_GetInfo((uint8*)pBuffer, nFileSize, &nWidth, &nHeight)) {
 				if (nWidth <= MAX_IMAGE_DIMENSION && nHeight <= MAX_IMAGE_DIMENSION) {
 					if ((double)nWidth * nHeight <= MAX_IMAGE_PIXELS) {
-						int nStride = Helpers::DoPadding(nWidth*3, 4);
+						int nStride = nWidth*4;

```

Comment:
you look like you know what you're doing and understand the concepts lol.  Can you help me to understand too?

Was this a bug before?  what does nStride do

Reply:
Both Webp_Dll_DecodeBGRInto() and Webp_Dll_DecodeBGRAInto() return data as an uint8 array.

Webp_Dll_DecodeBGRInto returned just R-G-B so the number of uint8 values in a line would be 3 times the width in pixels. For some reason or other, that value (nWidth*3) had to be padded to a multiple of 4, though, which was done with DoPadding().

Webp_Dll_DecodeBGRAInto() returns R-G-B-A, so the number of  uint8 values would be 4 times the width, and would always be a multiple of 4 anyway, so no padding is needed.

Reply:
thanks, that explanation helps!